services/api-server/src/simcore_service_api_server/api/routes/solvers_jobs_getters.py

```diff
@@ -389,7 +392,10 @@ async def get_job_wallet(
     _logger.debug("Getting wallet for job '%s'", job_name)
 
     if project_wallet := await webserver_api.get_project_wallet(project_id=job_id):
-        return await webserver_api.get_wallet(wallet_id=project_wallet.wallet_id)
+        return cast(
+            WalletGetWithAvailableCredits,
+            await webserver_api.get_wallet(wallet_id=project_wallet.wallet_id),
```

Comment:
Here I believe that the decorator ```_exception_mapper``` around webserver_api.get_wallet is obscuring the types.
In order to properly fix this you can do something similar as in https://github.com/itisfoundation/osparc-simcore/blob/e238d3e705ede10e497a3614890e3eb3d48388e7/packages/aws-library/src/aws_library/ssm/_error_handler.py that uses typing.ParamSpec and typing.TypeVar.
This makes both Pylance and mypy understand these decorated function. pretty cool stuff. almost feels like C++ templating

Reply:
Thx. Really nice input!